crates/database/src/orders.rs

```diff
@@ -108,6 +116,55 @@ impl Default for Order {
     }
 }
 
+pub async fn insert_pre_interactions(
+    ex: &mut PgConnection,
+    uid_and_preinteraction: &[(OrderUid, PreInteraction)],
+) -> Result<(), sqlx::Error> {
+    for (order_uid, pre_interaction) in uid_and_preinteraction.iter() {
+        match insert_pre_interaction(ex, pre_interaction, order_uid).await {
+            Ok(_) => (),
+            Err(err) if is_duplicate_record_error(&err) => (),
```

Comment:
This should no longer happen because the table doesn't have any UNIQUE constraint so we can remove the check here.